lib/ddtrace/profiling/pprof/converter.rb

```diff
@@ -84,6 +83,19 @@ def message
             "Mapping for sample value type '#{type}' to index is unknown."
           end
         end
+
+        protected
+
+        def new_group(event, values)
+          EventGroup.new(event, values)
+        end
+
+        def update_group(event_group, event, values)
+          # Update values for group
+          event_group.values.each_with_index do |group_value, i|
+            event_group.values[i] = group_value + values[i]
```

Comment:
I think there's a measurable performance difference for invoking `event_group.values` on every loop iteration, due to creating of the `values` arrays every time.
I believe the previous implementation, of storing the result of `event_group.values` would perform better here.

```suggestion
          group_values = event_group.values
          group_values.each_with_index do |group_value, i|
            group_values[i] = group_value + values[i]
```